apps/int32_publisher/app.c

```diff
@@ -1,61 +1,72 @@
-#include <allocators.h>
+#include <stdio.h>
 
 #include <rcl/rcl.h>
-#include <rcl_action/rcl_action.h>
 #include <rcl/error_handling.h>
 #include <std_msgs/msg/int32.h>
-#include "example_interfaces/srv/add_two_ints.h"
-
-#include <rmw_uros/options.h>
 
-#include <stdio.h>
-#include <unistd.h>
+#include <rclc/rclc.h>
+#include <rclc/executor.h>
 
-#define RCCHECK(fn) { rcl_ret_t temp_rc = fn; if((temp_rc != RCL_RET_OK)){printf("Failed status on line %d: %d. Aborting.\n",__LINE__,(int)temp_rc);  vTaskDelete(NULL);;}}
+#define RCCHECK(fn) { rcl_ret_t temp_rc = fn; if((temp_rc != RCL_RET_OK)){printf("Failed status on line %d: %d. Aborting.\n",__LINE__,(int)temp_rc);vTaskDelete(NULL);}}
 #define RCSOFTCHECK(fn) { rcl_ret_t temp_rc = fn; if((temp_rc != RCL_RET_OK)){printf("Failed status on line %d: %d. Continuing.\n",__LINE__,(int)temp_rc);}}
 
-void appMain(void *argument)
-{
-  rcl_init_options_t options = rcl_get_zero_initialized_init_options();
-
-  RCCHECK(rcl_init_options_init(&options, rcl_get_default_allocator()))
-
-  // Optional RMW configuration 
-  rmw_init_options_t* rmw_options = rcl_init_options_get_rmw_init_options(&options);
-  RCCHECK(rmw_uros_options_set_client_key(0xBA5EBA11, rmw_options))
```

Comment:
Just for my understanding: what does imply that this line is no longer present on the new code? Client key is automatically generated?

Reply:
Client key in micro-ROS can be set by using this (removed) functions if you want to set it at a certain value. But, by default it is set randomly when init the init_options.

Check this: https://github.com/micro-ROS/rmw-microxrcedds/blob/cee44c038c10f6f65268df07b1d677a7bb3697ff/rmw_microxrcedds_c/src/rmw_init.c#L78-L85

This function just replaces the random value set at the init.